commit 5a0438539cbe7a99e9b6d0ed21d7b73342b4a6f6
Author: Darrell Anderson <humanreadable@yahoo.com>
Date:   1336340005 -0500

    Update patch in GIT hash a9489034 to use reinterpret_cast<iCCP_data>.
    Thanks to /dev/ammo42.

diff --git a/digikam/libs/dimg/loaders/pngloader.cpp b/digikam/libs/dimg/loaders/pngloader.cpp
index fa6dad9..6862b6e 100644
--- a/digikam/libs/dimg/loaders/pngloader.cpp
+++ b/digikam/libs/dimg/loaders/pngloader.cpp
@@ -611,7 +611,7 @@ bool PNGLoader::save(const TQString& filePath, DImgLoaderObserver *observer)
     if (!profile_rawdata.isEmpty())
     {
 #if PNG_LIBPNG_VER_MAJOR > 1 || ( PNG_LIBPNG_VER_MAJOR == 1 && PNG_LIBPNG_VER_MINOR >= 5 )
-        png_set_iCCP(png_ptr, info_ptr, (png_charp)("icc"), PNG_COMPRESSION_TYPE_BASE, (iCCP_data)profile_rawdata.data(), profile_rawdata.size());
+        png_set_iCCP(png_ptr, info_ptr, (png_charp)("icc"), PNG_COMPRESSION_TYPE_BASE, reinterpret_cast<iCCP_data>(profile_rawdata.data()), profile_rawdata.size());
 #else
         png_set_iCCP(png_ptr, info_ptr, (png_charp)"icc", PNG_COMPRESSION_TYPE_BASE, profile_rawdata.data(), profile_rawdata.size());
 #endif
